iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_queue_popper.hpp

```diff
@@ -79,12 +79,10 @@ class ChunkQueuePopper
 
     /// @brief Attaches a condition variable
     /// @param[in] ConditionVariableDataPtr, pointer to an condition variable data object
-    /// @return True if successful, false if not
-    bool setConditionVariable(ConditionVariableData* conditionVariableDataPtr) noexcept;
+    void setConditionVariable(cxx::not_null<ConditionVariableData*> conditionVariableDataPtr) noexcept;
```

Comment:
Why not just use ``ConditionVariableData&`` ? ``not_null`` does runtime checks ...